quickjs/src/jsMain/kotlin/app/cash/quickjs/ZiplineJs.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.quickjs
+
+import app.cash.quickjs.internal.bridge.InboundService
+import app.cash.quickjs.internal.bridge.InternalBridge
+import app.cash.quickjs.internal.bridge.KtBridge
+import app.cash.quickjs.internal.bridge.OutboundClientFactory
+import kotlinx.coroutines.Dispatchers
+
+/**
+ * We declare this as a top-level property because that causes this object to be constructed when
+ * the code is loaded, and we rely on that for the side-effects performed in the ZiplineJs
+ * constructor.
+ */
```

Comment:
You can put side-effects into a `fun main()` which is what I did when I implemented QuickJs on its own companion. That allows elimination of this property and changing `ZiplineJs` to an `object` (or moving it to be directly implemented in `Zipline`'s `companion object`).

Reply:
I couldn’t figure out how to make this work. In particular I don’t know how to get a main() inside the zipline runtime library to load when it’s just a dependency of the testing project.

Reply:
`fun main` code should run when the library is loaded because it winds up being top-level in the JS closure. It _should_ behave same as the code which performs this field initialization.